core-proxy/src/main/java/eu/kennytv/maintenance/core/proxy/MaintenanceProxyPlugin.java

```diff
@@ -187,15 +189,35 @@ protected void kickPlayers() {
         kickPlayersFromProxy();
     }
 
+    @Blocking
+    @Nullable
     protected ProfileLookup doUUIDLookup(final String name) throws IOException {
-        final URL url = new URL("https://api.ashcon.app/mojang/v2/user/" + name);
-        final URLConnection connection = url.openConnection();
+        final URL url = new URL("https://api.mojang.com/users/profiles/minecraft/" + name);
+        final HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestMethod("GET");
+
+        int status = connection.getResponseCode();
+        if (status == 404) {
+            if (settingsProxy.isFallbackToOfflineUUID()) {
+                // Use offline uuid
+                return new ProfileLookup(UUID.nameUUIDFromBytes(("OfflinePlayer:" + name).getBytes(StandardCharsets.UTF_8)), name);
+            } else {
+                // Return null if profile not found
+                return null;
+            }
+        }
+
         try (final InputStream in = connection.getInputStream()) {
             final String output = CharStreams.toString(new InputStreamReader(in));
             final JsonObject json = GSON.fromJson(output, JsonObject.class);
 
-            final UUID uuid = UUID.fromString(json.getAsJsonPrimitive("uuid").getAsString());
-            final String username = json.getAsJsonPrimitive("username").getAsString();
+            final UUID uuid = UUID.fromString(
+                json.getAsJsonPrimitive("id").getAsString()
+                    .replaceFirst(
+                        "(\\p{XDigit}{8})(\\p{XDigit}{4})(\\p{XDigit}{4})(\\p{XDigit}{4})(\\p{XDigit}+)", "$1-$2-$3-$4-$5"
```

Comment:
This is a bit dank, I'd just have a more explicit method adding dashes at the required indexes

Reply:
Done

Reply:
As in, substring with the known indexes without a regex

Reply:
Alright, done